schemas/api_schema.js

```diff
@@ -0,0 +1,269 @@
+var { makeExecutableSchema } = require("graphql-tools");
+var FormatError = require("easygraphql-format-error");
+
+// Don't return more than MAX_SESSIONS_COUNT sessions in a single request!
+const MAX_SESSIONS_COUNT = 50;
+
+const errors = new FormatError([
+  {
+    name: "COUNT_TOO_HIGH",
+    message: "The passed count value is too high",
+    statusCode: 400,
+  },
+]).errorName;
+
+function strip_null(obj) {
+  Object.keys(obj).forEach((key) => obj[key] == null && delete obj[key]);
+  return obj;
+}
+
+// TODO: remove when login is implemented
+// TODO: all instances of user_token should be swapped with some real login auth token
+const USER_AUTHENTICATED = true;
+
+function check_auth(token) {
+  if (USER_AUTHENTICATED) {
+    // TODO: switch to the real user id
+    return "1";
+  }
+
+  throw new Error(errors.UNAUTHORIZED);
+}
+
+const schema = makeExecutableSchema({
+  typeDefs: [
+    `
+scalar Date
+
+type User {
```

Comment:
I don't think this is as significant as you think it is, why should I declare input types as well?

Reply:
This is useful is a specific series of input fields is recurring in many queries, since I don't think it's the case here yet it's ok to not use input types if you don't want to.